sim/global-types.ts

```diff
@@ -779,6 +779,7 @@ interface EventMethods {
 	onTypePriority?: number;
 }
 
+// FIXME: half of these fields do not belong here
```

Comment:
I will address this in a followup change as we discussed on Discord a while back